src/hb-ot-layout-gdef-table.hh

```diff
@@ -544,28 +577,67 @@ struct GDEF
 	   (version.to_int () >= 0x00010003u ? varStore.static_size : 0);
   }
 
+  void closure_variation_indices (hb_closure_variation_indices_context_t *c) const
+  { (this+ligCaretList).closure_variation_indices (c); }
+
+  void remap_layout_variation_indices (const hb_set_t *layout_variation_indices,
+				       hb_map_t *layout_variation_idx_map /* INOUT */) const
+  {
+    if (version.to_int () < 0x00010003u || !varStore) return;
+    if (layout_variation_indices->is_empty ()) return;
+    
+    hb_hashmap_t<uint16_t, hb_set_t *, 0xffff, nullptr> used_varidxes;
```

Comment:
There's a possibly simpler approach to this:
1. Create an iterator of hb_pair's with (major, minor).
2. Sort the iterator by major, then minor.
3. Filter out all pairs with major > varStore.dataSets.len
4. Then you can iterate through and do the remapping in one pass. Increment new major every time you encounter a new value for major.